ReleaseProcessScript.New.Tests/CommandLineGitClientTests.cs

```diff
@@ -0,0 +1,310 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using NUnit.Framework;
+using Optional;
+using Optional.Unsafe;
+
+namespace ReleaseProcessScript.New.Tests;
+
+[TestFixture]
+internal class CommandLineGitClientTests
+{
+  [SetUp]
+  public void Setup ()
+  {
+    _prevDir = Environment.CurrentDirectory;
+    var temp = Path.GetTempPath();
+
+    var guid = Guid.NewGuid();
+    var path = Path.Combine(temp, guid.ToString());
+    _remote = Directory.CreateDirectory(path).FullName;
+
+    Environment.CurrentDirectory = _remote;
+    _remoteName = "origin";
+
+    ExecuteGitCommand("--bare init");
+
+    guid = Guid.NewGuid();
+
+    path = Path.Combine(temp, guid.ToString());
+    _dir = Directory.CreateDirectory(path).FullName;
+
+    Environment.CurrentDirectory = _dir;
+
+    ExecuteGitCommand("init");
+    ExecuteGitCommand($"remote add {_remoteName} " + _remote);
+    ExecuteGitCommand("commit -m \"Initial Commit\" --allow-empty");
```

Comment:
Would it make sense to also push the commit here? I'm not sure as not all tests need it.